src/QueryObject.js

```diff
@@ -0,0 +1,62 @@
+import qs from "qs"
+import handleError from "./errors"
+
+class QueryObject {
+    constructor(client, endpoint) {
+
+        this.client = client
+        this.endpoint = endpoint
+
+        this.queryParams = {}
+        this.paginationParams = {}
+    }
+
+    where(queryParams = {}) {
+        const clonedQueryObject = Object.assign(new QueryObject(), this)
```

Comment:
I would suggest to remove `Object` because it's given anyway as JavaScript has objects natively.